clients/client/src/main/java/org/projectnessie/client/http/v2api/HttpCommitMultipleOperations.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.client.http.v2api;
+
+import org.projectnessie.client.builder.BaseCommitMultipleOperationsBuilder;
+import org.projectnessie.client.http.HttpClient;
+import org.projectnessie.error.NessieConflictException;
+import org.projectnessie.error.NessieNotFoundException;
+import org.projectnessie.model.Branch;
+import org.projectnessie.model.CommitResponse;
+import org.projectnessie.model.Reference;
+
+final class HttpCommitMultipleOperations extends BaseCommitMultipleOperationsBuilder {
+  private final HttpClient client;
+
+  HttpCommitMultipleOperations(HttpClient client) {
+    this.client = client;
+  }
+
+  @Override
+  public Branch commit() throws NessieNotFoundException, NessieConflictException {
```

Comment:
Since we're changing the API anyway, WDYT about wrapping these "bare" types into "per operation return types", e.g. some `CommitResult` type returned by this method? Similar to the wrapper types in the OpenAPI spec. (In a follow-up)

Reply:
So far all java client changes should be backward compatible, so older clients can switch to API v2 by just declaring `NessieApiV2.class` when they create the Nessie Client.

We could certainly add a _new_ commit variant to return the more complete result, e.g. `commitAdvanced()`, WDYT?

Reply:
Yea - just raised this. Can you open a follow-up issue?

Reply:
#5487